packages/analytics/dashboard-renderer/src/components/DashboardRenderer.cy.ts

```diff
@@ -1,30 +1,75 @@
-// Cypress component test spec file
-
-import DashboardRenderer from './DashboardRenderer.vue'
 import { ChartTypes } from '../types'
 import { ChartMetricDisplay } from '@kong-ui-public/analytics-chart'
-import type { GlobalMountOptions } from '@vue/test-utils/dist/types'
-import { nonTsExploreResponse } from '../../sandbox/mock-data'
-
 import { INJECT_QUERY_PROVIDER } from '../constants'
-import type { AnalyticsBridge } from '@kong-ui-public/analytics-utilities'
+import type { AnalyticsBridge, ExploreQuery, ExploreResultV4, Timeframe } from '@kong-ui-public/analytics-utilities'
+import { generateSingleMetricTimeSeriesData, generateCrossSectionalData, TimeframeKeys, TimePeriods } from '@kong-ui-public/analytics-utilities'
+import DashboardRenderer from './DashboardRenderer.vue'
+import { nonTsExploreResponse, timeSeriesExploreResponse, summaryDashboardConfig } from '../../sandbox/mock-data'
 
-const mockQueryProvider: AnalyticsBridge = {
-  queryFn: async () => { return nonTsExploreResponse },
-}
+describe('<DashboardRenderer />', () => {
+  beforeEach(() => {
+    cy.viewport(1200, 1000)
+  })
 
-const global: GlobalMountOptions = {
-  provide: {
-    [INJECT_QUERY_PROVIDER]: mockQueryProvider,
-  },
-}
+  const mockQueryProvider = (): AnalyticsBridge => {
+    const queryFn = (query: ExploreQuery): Promise<ExploreResultV4> => {
+      // Dimensions to use if query is not provided
+      const dimensionMap = { statusCode: ['1XX', '2XX', '3XX', '4XX', '5XX'] }
+
+      if (query.dimensions && query.dimensions.findIndex(d => d === 'time') > -1) {
+        if (query.metrics && query.metrics[0] === 'request_count') {
+        // Traffic + Error rate cards
+          return Promise.resolve(nonTsExploreResponse)
+        } else if (query.metrics && query.metrics[0] === 'response_latency_p99' && query.metrics.length === 1) {
+        // Latency metrics card
+          return Promise.resolve(timeSeriesExploreResponse)
+        } else {
+        // Timeseries Line chart
+          const timeSeriesResponse = generateSingleMetricTimeSeriesData(
+            { name: 'TotalRequests', unit: 'count' },
+            { statusCode: query.metrics as string[] },
+          ) as ExploreResultV4
+
+          return Promise.resolve(timeSeriesResponse)
+        }
+      } else {
+      // Bar charts (non-time series)
+        const nonTimeSeriesResponse = generateCrossSectionalData(
+          [
+            { name: 'TotalRequests', unit: 'count' },
+          ],
+          dimensionMap,
+        ) as ExploreResultV4
+
+        return Promise.resolve(nonTimeSeriesResponse)
+      }
+    }
+
+    return {
+      queryFn: cy.spy(queryFn).as('fetcher'),
+    }
+  }
+
+  afterEach(() => {
```

Comment:
Could you add a code comment explaining why this is here?

Reply:
The first test in the file loads up in a very weird manner, and this injected CSS block was my attempt to style the Cypress component runner in a way that allows both charts to show inside the viewport.  But since the subsequent tests look good (weirdly enough), I should probably just remove it.

<img width="1110" alt="Screenshot 2024-03-06 at 12 37 10" src="https://github.com/Kong/public-ui-components/assets/103061463/227954a5-56ab-4294-8ae8-39f9fdf3af28">